engines/sherlock/animation.cpp

```diff
@@ -0,0 +1,208 @@
+				pt.x = stream->readUint16LE();
+				pt.y = stream->readUint16LE();
+			} else {
+				pt = images[imageFrame]._offset;
+			}
+
+			// Draw the sprite. Note that we explicitly use the raw frame below, rather than the ImageFrame,
+			// since we don't want the offsets in the image file to be used, just the explicit position we specify
+			screen.transBlitFrom(images[imageFrame]._frame, pt);
+		} else {
+			// At this point, either the sprites for the frame has been complete, or there weren't any sprites
+			// at all to draw for the frame
+			if (fade == 255) {
+				// Gradual fade in
+				if (screen.equalizePalette(images._palette) == 0)
+					fade = 0;
+			}
+
+			// Check if we've reached a frame with sound
+			if (frameNumber++ == *soundFrames) {
+				++soundNumber;
+				++soundFrames;
+				Common::String fname = _vm->_soundOverride.empty() ?
+					Common::String::format("%s%01d", filename.c_str(), soundNumber) :
+					Common::String::format("%s%02d", filename.c_str(), soundNumber);
+
+				if (sound._voices)
+					sound.playSound(fname, WAIT_RETURN_IMMEDIATELY);
+			}
+
+			events.wait(speed * 3);
+		}
+
+		if (events.kbHit()) {
+			Common::KeyState keyState = events.getKey();
+			if (keyState.keycode == Common::KEYCODE_ESCAPE ||
+				keyState.keycode == Common::KEYCODE_SPACE) {
+				skipped = true;
+				break;
+			}
+		} else if (events._pressed) {
+			skipped = true;
+			break;
+		}
+	}
+
+	events.clearEvents();
+	sound.stopSound();
+	delete stream;
+	sound._playingEpilogue = false;
+
+	return !skipped && !_vm->shouldQuit();
+}
+
+/**
+ * Load the prologue name array 
+ */
+void Animation::setPrologueNames(const char *const *names, int count) {
+	for (int idx = 0; idx < count; ++idx, names++) {
+		_prologueNames.push_back(*names);
+	}
+}
+
+/**
+ * Load the prologue frame array
+ */
+void Animation::setPrologueFrames(const int *frames, int count, int maxFrames) {
+	_prologueFrames.resize(count);
+
+	for (int idx = 0; idx < count; ++idx, frames += maxFrames) {
+		_prologueFrames[idx].resize(maxFrames);
+		Common::copy(frames, frames + maxFrames, &_prologueFrames[idx][0]);
+	}
+}
+
+/**
+ * Load the title name array
+ */
+void Animation::setTitleNames(const char *const *names, int count) {
+	for (int idx = 0; idx < count; ++idx, names++) {
+		_titleNames.push_back(*names);
+	}
+}
+
+/**
+ * Load the title frame array
+ */
+void Animation::setTitleFrames(const int *frames, int count, int maxFrames) {
+	_titleFrames.resize(count);
+
+	for (int idx = 0; idx < count; ++idx, frames += maxFrames) {
+		_titleFrames[idx].resize(maxFrames);
+		Common::copy(frames, frames + maxFrames, &_titleFrames[idx][0]);
+	}
+}
+
+/**
+ * Checks for whether an animation is being played that has associated sound
+ */
+const int *Animation::checkForSoundFrames(const Common::String &filename) {
+	const int *frames = &NO_FRAMES;
+
+	if (_vm->_soundOverride.empty()) {
+		for (Common::Array<const char *>::size_type idx = 0; idx < _prologueNames.size(); ++idx) {
```

Comment:
I don't really see the point for using size_type, now that you point it out, rather just using uint